### ISSUE TITLE
Do you intend to open your code?

### ISSUE THREAD

**Issue by lyrgwlr** [wmcnally/golfdb#2]
I'm interesting on your paper:GolfDB:AVideoDatabaseforGolfSwingSequencing. 
So I want to request your SwingNet code for research purpose.
Thanks a lot.

**Comment by lyrgwlr** [comment]
What's the 'bbox' column exactly means in the dataframe you provided.
In the paper, it's described as:
"The bounding boxes were drawn to include the clubhead and golf ball through the full duration of the swing."
But it seems only 1x4 array is provided.

**Comment by wmcnally** [comment]
There is only one bounding box for each video. The bounding box was drawn such that the clubhead and golf ball were visible in all frames. I will be uploading some code soon. I am working on re-implementing the model in TF.

**Comment by lyrgwlr** [comment]
@wmcnally 
Hi, I have a little confuse about your paper.
In 6.3 Baseline SwingNet, you said the batch size is 24, and the sequence length is 64. 
So the input's shape is (24*64, 3, 160, 160) , is that right?
I have tried load the tensor into my NVIDIA 1080Ti which has 11G memory, but out of memory happended.
Would you help me? Thanks a lot.

**Comment by wmcnally** [comment]
That’s the same input shape I used. Recall in the paper I mentioned that I froze the first 10 layers of MobileNetV2. I was also using a Titan X (12GB memory). So you will need to lower your batch size. I will upload some PyTorch code this week.

**Comment by lyrgwlr** [comment]
@wmcnally ,
Probably something goes wrong. Because I have tried frozening the first 10 layers of MobileNetV2, and I set my batch size to even 10, the out of memory errors still came...
So I'm looking forward for your code. Thanks~

**Comment by wmcnally** [comment]
@lyrgwlr I uploaded some starter PyTorch code and a link to preprocessed videos. Please let me know if there are any issues.

**Comment by lyrgwlr** [comment]
@wmcnally,
I have a question about the baseline model.
The paper mentions that "the baseline SwingNet was trained for 7k iterations on each split of GolfDB, and the learning rate was reduced by an order of magnitude after 5k iterations. "
Is that means that total epoches are 7k*5 = 35k? and the lr was reduced 7 times? 
or the initial lr=0.001 and reduced to 0.0001 after 5k iterations on each spilt?

**Comment by lyrgwlr** [comment]
@wmcnally , 
Thanks for your code, and the training can run well.
I add some codes:
1. random affine and horizontal flip as the paper said.
2. xaiver normal/uniform initialization for the fc layer.

And my batch size is 22, sequence length is 64. I trained 4 splits and each for 7k iterations. The model is loaded once before the training and the learning rate I set is like this. Because the paper said "the learning rate was reduced by an order of magnitude after 5k iterations":
![image](https://user-images.githubusercontent.com/25834598/70884656-9d12a880-2011-11ea-945e-b663b88f0110.png)

(I also tried setting the learning rate like this:
![image](https://user-images.githubusercontent.com/25834598/70884825-0b576b00-2012-11ea-88ec-18ea4673db3d.png)
The result is wrose about 5% PCE. )

Then the PCE of 4 splits is like this:
![image](https://user-images.githubusercontent.com/25834598/70884916-45c10800-2012-11ea-9a93-4af9e9a05469.png)

Obviously it's far from the paper. So are there something I missed?

**Comment by wmcnally** [comment]
@lyrgwlr 
It looks like you trained a single model for 28k iterations. This is incorrect. Are you familiar with k-fold cross-validation? You should do a 4-fold cross validation, where you train 4 separate models (1 for each split). In the paper I used a learning rate of 0.001 for the first 5k iterations and 0.0001 for the remaining 2k iterations. Save checkpoints frequently as overfitting may be a reason for low PCE. Depending on your implementation of random affine transformations you may get different results than in the paper. The paper also used a batch size of 24.

**Comment by lyrgwlr** [comment]
@wmcnally ,
sorry for bother you.  :(
Is the 4-fold cross validation do it like that:
1. For each split, create a new model and train 7k iterations.
2. Then run eval.py for each split using corresponding checkpoints of each model.

I trained like this today but still get the bad results (about 42% PCE).

**Comment by wmcnally** [comment]
@lyrgwlr
There could be something wrong with your data augmentation. My guess is you are overfitting. What is the PCE for the same model after 2k iterations?

**Comment by lyrgwlr** [comment]
@wmcnally 
The PCE for the same model after 2k iterations is little lower than 7k iterations so I think it's no the overfitting issue.
Besides, here are some images after my data augmentation. I did 50% percent horizontal flip and random affine with +-5° rotation and  +-5° shear.
![413](https://user-images.githubusercontent.com/25834598/71048332-0a414d80-217a-11ea-9944-38c0487934d5.jpg)
![571](https://user-images.githubusercontent.com/25834598/71048361-15947900-217a-11ea-80e1-99255b395ef8.jpg)
![629](https://user-images.githubusercontent.com/25834598/71048393-2513c200-217a-11ea-8df1-1b73d1aa0a3f.jpg)
![0](https://user-images.githubusercontent.com/25834598/71048408-2e049380-217a-11ea-8b6c-789994715661.jpg)

Please let me know if something wrong with my data augmentation. Today I will train without horizontal flip and random affine and report the result to you. Thanks~

**Comment by wmcnally** [comment]
@lyrgwlr 
The training code I provided gives a PCE of 71.5 after ~1.8k iterations so there could be something wrong with your code. If you implement data augmentation correctly you should be able to improve the PCE. I'm closing this issue as there is nothing wrong with the code I have provided. Good luck!

**Comment by lyrgwlr** [comment]
@wmcnally ,
I figured it out. It's the FPS problem. The videos I downloaded from Youtube are not all the 30 FPS but some videos are 60FPS. So I made some changes in "preprocess_videos.py". But I forgot to change the codes in "dataloader.py". 
events *= round(fps/30) is needed if the videos are not the 30FPS otherwise the labels can't match the real video files.
I got the normal PCE result now.
I hope this could help those who have the similar problem like I met.
